File: reto_clean_code/angular/step2-discount.ts
// Función para calcular descuentos según el tipo de cliente.  Se utiliza en
// el paso 2 para practicar TDD y refactorización.

export function calcularDescuento(precio: number, tipoCliente: string): number {
  const descuentos: { [key: string]: number } = {
    preferente: 0.1,
    vip: 0.2
  };
  return precio * (1 - (descuentos[tipoCliente] || 0));
}

File: reto_clean_code/python/step2_discount.py
"""
Módulo para calcular descuentos según el tipo de cliente.  Esta función se
utiliza en el paso 2 para practicar TDD y refactorización.
"""


def calcular_descuento(precio: float, tipo_cliente: str) -> float:
    descuentos = {
        'preferente': 0.1,
        'vip': 0.2,
    }
    return precio * (1 - descuentos.get(tipo_cliente, 0))